docs/basics/DeepRLTutorial/TutorialDeepRL.rst

```diff
@@ -0,0 +1,605 @@
+parameters w and depending on the algorithm it could be action-value
+:math:`Q_{\varphi}(s,a )` or state-value :math:`V_{\varphi}(s)`. - Actor
+updates the policy parameters :math:`\theta` for
+:math:`\pi_{\theta}(a \mid s)`, in the direction suggested by the
+critic.
+
+**A2C** is an Actor-Critic algorithm and it is part of the on-policy
+family, which means that we are learning the value function for one
+policy while following it. The original paper in which it was proposedi
+can be found `here <https://arxiv.org/pdf/1602.01783.pdf>`__ and the
+pseudocode of the algorithm is the following:
+
+-  Initialize the actor :math:`\pi_{\theta}` and the critic
+   :math:`V_{\varphi}` with random weights.
+-  Observe the initial state :math:`s_{0}`.
+-  for :math:`t \in\left[0, T_{\text {total }}\right]` :
+
+   -  Initialize empty episode minibatch.
+   -  for :math:`k \in[0, n]:` # Sample episode
+
+      -  Select a action :math:`a_{k}` using the actor
+         :math:`\pi_{\theta}`.
+      -  Perform the action :math:`a_{k}` and observe the next state
+         :math:`s_{k+1}` and the reward :math:`r_{k+1}`.
+      -  Store :math:`\left(s_{k}, a_{k}, r_{k+1}\right)` in the episode
+         minibatch.
+
+   -  if :math:`s_{n}` is not terminal: set
+      :math:`R=V_{\varphi}\left(s_{n}\right)` with the critic, else
+      :math:`R=0`.
+   -  Reset gradient :math:`d \theta` and :math:`d \varphi` to 0 .
+   -  for :math:`k \in[n-1,0]` : # Backwards iteration over the episode
+
+      -  Update the discounted sum of rewards
+         :math:`R \leftarrow r_{k}+\gamma R`
+      -  Accumulate the policy gradient using the critic:
+
+         .. math::
+
+
+            d \theta \leftarrow d \theta+\nabla_{\theta} \log \pi_{\theta}\left(a_{k}\mid s_{k}\right)\left(R-V_{\varphi}\left(s_{k}\right)\right)
+
+      -  Accumulate the critic gradient:
+
+.. math::
+
+
+   d \varphi \leftarrow d \varphi+\nabla_{\varphi}\left(R-V_{\varphi}\left(s_{k}\right)\right)^{2}
+
+-  Update the actor and the critic with the accumulated gradients using
+   gradient descent or similar:
+
+.. math::
+
+
+   \theta \leftarrow \theta+\eta d \theta \quad \varphi \leftarrow \varphi+\eta d \varphi
+
+Running A2C on CartPole
+-----------------------
+
+In the next example we use default parameters for both the Actor and the
+Critic and we use rlberry to train and evaluate our A2C agent. The
+default networks are:
+
+-  a multilayer perceptron with two hidden layers of 64 units for the
+   **Actor**, the input layer has the dimension of the state space while
+   the output layer has the dimension of the action space. The
+   activations are RELU functions and we have a softmax in the last
+   layer.
+-  a multilayer perceptron with two hidden layers of 64 units for the
+   **Critic**, the input layer has the dimension of the state space
+   while the output has dimension 1. The activations are RELU functions
+   apart from the last layer that has a linear activation.
```

Comment:
Minor detail, but I think people are referring to MLPs more as dense neural networks nowadays.